### PR TITLE
Deliver precompiled libmcljava.so, mcljava.dll etc.

### PR DESCRIPTION
**Pull request by WorldofJARcraft**
This PR introduces bundling native libraries for Windows, Linux and OSX into the assembled JAR artefact, aiming at making installation easier in non-performance-critical environments (e.g. CI pipelines), as discussed in #25.
Included changes:
  - pre-compiled native libraries for Windows (x86 and x64), Linux (x86 and x64), and Mac (x64) included
  - scripts for building portable libraries on Windows (no GMP dependency, compatible with all x86 and x64 CPUs) and Linux (no GMP, libgcc, libstdc++ dependency, compatible with all x86 and x64 CPUs)
  - pre-compiled Mac library. It was compiled using the standard "install_mcl.sh" script. Since I do not own a Mac, I could not experiment with compiler and linker flags here. Also, I could not provide a M1 binary for the same reason. However, using the existing scripts the respective binary should be easy to add.
  - I moved the compile scripts into a "scripts" subdirectory and renamed them. This breaks existing scripts that assume an install_mcl.sh to exist in the top level directory, so feel free to revert this change.
  - Java determines the operating system and architecture and loads the included libraries if no mcljava library was found in the library path.

### CODE REVIEW

**Comment by feidens** [comment]
PR looks really good to me. Thank you.
- "I moved the compile scripts into a "scripts" subdirectory and renamed them. This breaks existing scripts that assume an install_mcl.sh to exist in the top level directory, so feel free to revert this change." This path change could be of interest for @this-kramer and @pschuermann97.

File: src/main/java/org/cryptimeleon/mclwrap/bn254/MclBilinearGroupImpl.java [hunk 1]
```diff
@@ -44,19 +49,96 @@ public static boolean isAvailable() {
...
+        String tempdir;
+        try {
+            tempdir = Files.createTempDirectory("mclwrap").toAbsolutePath().toString();
+        } catch (IOException e) {
+            return false;
+        }
+
+        File outputFile = new File(tempdir+File.separator+requiredLibraryName);
+
+        if(!outputFile.exists()){
```
Comment (JanBobolz) [inline src/main/java/org/cryptimeleon/mclwrap/bn254/MclBilinearGroupImpl.java]: I'd vote to replace this with `if(true)`, essentially. I don't see the user providing a library by writing it into a tmp dir.
On the flip side, this if may have weird consequences when, for example, we upgrade mcl binaries at some point and someone still has the old ones in the tmp dir.

**Review by JanBobolz — CHANGES_REQUESTED** [review]
Great work! Just a small change request, then this can be merged.

Thanks a lot :)

**Comment by JanBobolz** [comment]
Oh. The script move should also break the github workflows. Weird that this is not being reported by CI already. 

Anyway, we can fix that as soon as the merge goes through.

File: src/main/java/org/cryptimeleon/mclwrap/bn254/MclBilinearGroupImpl.java (hunk 1 above)
Comment (WorldofJARcraft) [reply src/main/java/org/cryptimeleon/mclwrap/bn254/MclBilinearGroupImpl.java]: Done.

**Review by JanBobolz — APPROVED** [review]

**Pull request merged**